### ISSUE TITLE
executeSelectPage() is not working as expected

### ISSUE THREAD

**Issue by NicolasRoehm** [juanluispaz/ts-sql-query#69]
Hi Juan,

Thank you very much for the great lib, I love it!

I have a lot of records in this table, but when I'm using `executeSelectPage()` with `limit(10)` and `offset(0)` I'm getting `count : 10` instead of 375.

![image](https://user-images.githubusercontent.com/6847846/207724038-43ccf555-941d-4130-8fb2-ba381bfd3de3.png)

In the documentation, it says : *Select page execute the query twice, the first one to get the data from the database and the second one to get the count of all data without the limit and the offset.*

Best regards,
Nicolas

**Comment by juanluispaz** [comment]
Can you see the executed SQL? I already have lot of tests over this and I'm not able to figure out how it can be happening.

You can write the executed SQL using: https://ts-sql-query.readthedocs.io/en/stable/query-runners/general-purpose-query-runners/#consolelogqueryrunner

I will be interested if you can share the sql of the first query and the second one as well; and if you can share the code that generates the query I will appreciate

Thanks.

**Comment by NicolasRoehm** [comment]
Executed SQL 

```sql
selectOneRow select id as id, created as created, created_by as "createdBy", updated as updated, updated_by as "updatedBy", 
deleted as deleted, deleted_by as "deletedBy", email as email, password_hash as "passwordHash", first_name as "firstName", last_name as "lastName", sso_ship_tos as "ssoShipTos", email_verified as "emailVerified", role as role, email_change_token as "emailChangeToken", email_verification_token as "emailVerificationToken", password_reset_token as "passwordResetToken" from "user" where id = $1 and deleted is null [ '0f6a673e-f36b-1410-8b19-000c67a90e03' ]

selectManyRows select "user".id as id, "user".email as email, "user".first_name as "firstName", "user".last_name as "lastName", "user".role as role, json_agg(json_build_object('id', establishment.id, 'businessCode', establishment.business_code, 'name', establishment.name)) as establishments from "user" left join establishment_user as eu on "user".id = eu.user_id left join establishment as establishment on eu.establishment_id = establishment.id and "user".deleted is null group by "user".id limit $1 offset $2 [ 10, 10 ]

selectOneColumnOneRow with result_for_count as (select "user".id as id, "user".email as email, "user".first_name as firstName, "user".last_name as lastName, "user".role as role, json_agg(json_build_object('id', establishment.id, 'businessCode', establishment.business_code, 'name', establishment.name)) as establishments from "user" left join establishment_user as eu on "user".id = eu.user_id left join establishment as establishment on eu.establishment_id = establishment.id and "user".deleted 
is null group by "user".id limit $1 offset $2) select count(*) from result_for_count [ 10, 10 ]
```

TypeScript source code
```ts
const establishment     = tEstablishment.forUseInLeftJoinAs('establishment');
const establishmentUser = tEstablishmentUser.forUseInLeftJoinAs('eu');
const userPage = await this.connection.selectFrom(tUser)
  .leftJoin(establishmentUser).on(tUser.id.equals(establishmentUser.userId))
  .leftJoin(establishment).on(establishmentUser.establishmentId.equals(establishment.id))
  .and(tUser.deleted.isNull())
  .groupBy(tUser.id)
  .select({
    id        : tUser.id,
    email     : tUser.email,
    firstName : tUser.firstName,
    lastName  : tUser.lastName,
    role      : tUser.role,
    establishments : this.db.aggregateAsArray({
      id           : establishment.id,
      businessCode : establishment.businessCode,
      name         : establishment.name,
    }).useEmptyArrayForNoValue()
  })
  .limit(10)
  .offset(0)
  .executeSelectPage();
```
Models (ManyToMany)

```User -> EstablishmentUser <- Establishment```

Typed result
```ts
const userPage: {
  data: {
    email: string;
    firstName: string;
    lastName: string;
    role: number;
    id: string;
    establishments: {
      id: string;
      businessCode: string;
      name: string;
    }[];
  }[];
  count: number;
}
```

**Comment by juanluispaz** [comment]
Thanks for the additional information, with this I was able to reproduce the issue and fix it in ts-sql-query 1.47.0 already released.

I'm happy you love the work I put here. Please, don't doubt to contact me if you find an issue, you have a wish that maybe I can include in the library, or if you have any question; I will be pleased to answer you.
